game-core/src/main/java/games/strategy/triplea/delegate/AbstractEndTurnDelegate.java

```diff
@@ -516,7 +517,7 @@ public String getDisplayName() {
   private static Comparator<Territory> getSingleNeighborBlockadesThenHighestToLowestProduction(
       final Collection<Territory> blockadeZones, final GameMap map) {
     return (t1, t2) -> {
-      if (t1 == t2 || (t1 == null && t2 == null)) {
+      if (ObjectUtils.referenceEquals(t1, t2)) {
```

Comment:
A few things to note here:

1. The second expression, `t1 == null && t2 == null`, is redundant; `t1 == t2` already accounts for this condition.  Therefore the `referenceEquals()` call is sufficient.
1. I considered replacing this expression with simply `Objects.equals(t1, t2)` and deleting the test of `t1.equals(t2)` a few lines below; the resulting code should be equivalent.  However, I didn't because
    1. there's zero test coverage in this method, and we've been burned with changing comparators without tests in the past, and
    1. I'm not sure if someone tried to optimize this in the past for the `null` cases (are `null` territories even a thing to worry about here?).

I could be compelled on the second point to use `Objects#equals()` with the right amount of nudging. :smile:

(Ditto for the same change in the following method.)

Reply:
`Objects#equals` :P

Reply:
fef1a126e